pkg/icingadb/db.go

```diff
@@ -89,12 +89,33 @@ func (db *DB) BuildInsertStmt(into interface{}) (string, int) {
 	), len(columns)
 }
 
-func (db *DB) BuildSelectStmt(from interface{}, into interface{}) string {
+// BuildInsertIgnoreStmt returns an INSERT statement for the specified struct for
+// which the database ignores rows that have already been inserted.
+func (db *DB) BuildInsertIgnoreStmt(into interface{}) (string, int) {
+	columns := db.BuildColumns(into)
+
 	return fmt.Sprintf(
+		// MySQL treats UPDATE id = id as a no-op.
+		`INSERT INTO %s (%s) VALUES (%s) ON DUPLICATE KEY UPDATE id = id`,
+		utils.TableName(into),
+		strings.Join(columns, ", "),
+		fmt.Sprintf(":%s", strings.Join(columns, ", :")),
+	), len(columns)
+}
+
+func (db *DB) BuildSelectStmt(from interface{}, into interface{}) string {
+	q := fmt.Sprintf(
 		`SELECT %s FROM %s`,
 		strings.Join(db.BuildColumns(into), ", "),
 		utils.TableName(from),
 	)
+
+	if scoper, ok := from.(contracts.Scoper); ok {
+		where, _ := db.BuildWhere(scoper.Scope())
+		q = q + ` WHERE ` + where
```

Comment:
Btw. I'd let every of my trainees dig through half of Google on how to shorten this line by three characters... but I'm afraid you're my boss. 🤷‍♂️

Reply:
Hmm, let me guess! ;)
```go
wh, _ := db.BuildWhere(scoper.Scope())
q = q + ` WHERE ` + wh
```

Reply:
Not the three characters I meant :)

Reply:
> but I'm afraid you're my boss. 🤷‍♂️

But I also want to learn, so what line are we talking about exactly?

Reply:
115

Reply:
Ok, I have no clue! 😆 Please help!

Reply:
If there only was a way to append to a string without typing the variable name twice!

Reply:
🐘